libs/utils/test.py

```diff
@@ -213,16 +213,25 @@ def experiment_test(wrapped_test, instance, args, kwargs):
     The method will be passed the experiment object and a list of the names of
     tasks that were run as the experiment's workload.
     """
+    results = {}
     for experiment in instance.executor.experiments:
         tasks = experiment.wload.tasks.keys()
+        tag = '{}_{}'.format(experiment.wload_name, experiment.conf)
+        results[tag] = results.get(tag, [])
         try:
             wrapped_test(experiment, tasks, *args, **kwargs)
         except AssertionError as e:
             trace_relpath = os.path.join(experiment.out_dir, "trace.dat")
-            add_msg = "\n\tCheck trace file: " + os.path.abspath(trace_relpath)
-            orig_msg = e.args[0] if len(e.args) else ""
-            e.args = (orig_msg + add_msg,) + e.args[1:]
-            raise
+            add_msg = "Check trace file: " + os.path.abspath(trace_relpath)
+            results[tag].append(str(e) + "\n\t" +  add_msg)
+        else:
+            results[tag].append(None)
```

Comment:
I guess this can be required to account for how many tests we executed for a certain wload/configurations... but is not that number exactly the configured number of iterations?